src/backend/opencl/blas.cpp

```diff
@@ -113,6 +118,10 @@ template<typename T>
 Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs,
                 af_mat_prop optLhs, af_mat_prop optRhs)
 {
+    if(OpenCLCPUOffload()) {
```

Comment:
The behavior is also off by default. Users will enable it using a flag. We can document the flag saying that it can cause synchronization. The "mapping" operation itself is synchronous by the way. It would require too much overhead with very little payback to make this operation asynchronous.